packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'actions.dart';
+import 'framework.dart';
+
+/// A set of [LogicalKeyboardKey]s that can be used as the keys in a map.
+///
+/// This is mainly used by [ShortcutMap] to allow definition of shortcut
+/// mappings.
+class LogicalKeySet {
+  /// A constructor for making a [LogicalKeySet] of up to four keys.
+  ///
+  /// If you need a set of more than four keys, use [LogicalKeySet.fromSet].
+  ///
+  /// The `key1` parameter must not be null.
+  LogicalKeySet(
+    LogicalKeyboardKey key1, [
+    LogicalKeyboardKey key2,
+    LogicalKeyboardKey key3,
+    LogicalKeyboardKey key4,
+  ])  : assert(key1 != null),
+        _logicalSet = <LogicalKeyboardKey>{} {
+    _logicalSet.add(key1);
+    if (key2 != null) {
+      _logicalSet.add(key2);
+    }
+    if (key3 != null) {
+      _logicalSet.add(key3);
+    }
+    if (key4 != null) {
+      _logicalSet.add(key4);
+    }
+  }
+
+  /// Create  a [LogicalKeySet] from a set of [LogicalKeyboardKey]s.
+  ///
+  /// The `keys` must not be null or empty.
+  LogicalKeySet.fromSet(Set<LogicalKeyboardKey> keys)
+      : assert(keys != null),
+        assert(keys.isNotEmpty),
+        _logicalSet = keys;
+
+  final Set<LogicalKeyboardKey> _logicalSet;
+
+  /// Convert the [LogicalKeySet] to a set of [LogicalKeyboardKey]s and return a
+  /// copy.
+  Set<LogicalKeyboardKey> toSet() => _logicalSet.toSet();
+
+  @override
+  bool operator ==(dynamic other) {
```

Comment:
see style guide for a template for operator ==

Reply:
Fixed. FYI, I did notice that almost all of our `operator ==` in the tree use dynamic and not Object.

Reply:
Yeah, at some point we should really do an audit...